ember.s

```diff
@@ -369,14 +377,14 @@ do_disk:
 	int i_biosdisk
 	jc short error
 
-	pop ds
+.ret:	pop ds
 	popa
 	iret
 
 error:
 	mov al, '!'
 	int i_putchar
-	jmp shell
+	jmp short do_disk.ret
```

Comment:
I don't think this is a good solution either. One of the reasons these are exposed as interrupts is to let programs ran with ember use these functions. Returning like that will lead them to believe that the reads or writes worked, when they didn't.